heaps/min_heap.py

```diff
@@ -21,19 +21,28 @@ def __init__(self):
     def add(self, key, value = None):
         """ This method adds a HeapNode instance to the heap
             If value == None the new node's value should be set to key
-            Time Complexity: ?
-            Space Complexity: ?
+            Time Complexity: O log n
+            Space Complexity: O 1
         """
-        pass
+        if not value:
+            value = key
+
+        node = HeapNode(key, value)
+        self.store.append(node)
+        self.heap_up(len(self.store)-1)
 
     def remove(self):
         """ This method removes and returns an element from the heap
             maintaining the heap structure
-            Time Complexity: ?
-            Space Complexity: ?
+            Time Complexity: O log n
+            Space Complexity: O 1
         """
```

Comment:
👍 , However the space complexity is O(log n) because of `heap_down`'s recursive call stack